lib/Backend/Lower.cpp

```diff
@@ -29161,3 +29109,198 @@ Lowerer::LowerCheckUpperIntBound(IR::Instr * instr)
     return instrPrev;
 }
 #endif
+
+Lowerer::LowerGeneratorHelper::LowerGeneratorHelper(Func* func, Lowerer* lowerer, LowererMD& lowererMD):
+    func(func), lowerer(lowerer), lowererMD(lowererMD) {}
+
+void Lowerer::LowerGeneratorHelper::InsertNullOutGeneratorFrameInEpilogue(IR::LabelInstr* epilogueLabel)
+{
+    IR::Instr* insertionPoint = epilogueLabel->m_next;
+
+    // null out frame pointer on generator object to signal completion to JavascriptGenerator::CallGenerator
+    // s = MOV prm1
+    // s[offset of JavascriptGenerator::frame] = MOV nullptr
+    StackSym* symSrc = StackSym::NewImplicitParamSym(3, this->func);
+    this->func->SetArgOffset(symSrc, LowererMD::GetFormalParamOffset() * MachPtr);
+    IR::SymOpnd* srcOpnd = IR::SymOpnd::New(symSrc, TyMachPtr, this->func);
+    IR::RegOpnd* dstOpnd = IR::RegOpnd::New(TyMachReg, this->func);
+
+    // We would replace the values in all of the callee-saved registers
+    // in the epilogue anyway, so use one of them to store the address
+    // to our interpreter frame
+#if defined(_M_X64)
+    dstOpnd->SetReg(RegR15); // R15 is callee-save on both Windows/Linux
```

Comment:
do we save/restore all callee save registers for generators? normally we only save/restore the ones we use

Reply:
Oh so it seems I made a wrong assumption. I always thought we save/restore all callee-saved registers described in `RegList.h`. Is there a register that we always use that I can use for this purpose then?

Reply:
I think what you actually want is to use a caller-save register. The epilogue code doesn't use /need values in any of them. You could pick whatever you want except for the return register (RAX), so ECX/RCX would be a good choice.